docs/pages/developer-guide.md

```diff
@@ -936,6 +936,160 @@ In the example above, a server task can be accessed through the address:
 server-lb.hello-world.l4lb.thisdcos.directory:80
 ```
 
+# TLS
+
+**This feature works only on `Mesosphere DC/OS Enterprise` and is not supported on `DC/OS open`.**
+
+The SDK provides automated way of provisioning X.509 certificates and private keys referered in this documentation also as TLS artifacts. TLS artifacts can be used to expose service over encrypted TLS connection.
+
+One or more TLS artifacts can be requrested by adding `transport-encryption` key to the `task` YAML definition.
+
+```yaml
+name: "hello-world"
+pods:
+  hello:
+    count: 1
+    tasks:
+      server:
+        goal: RUNNING
+        cmd: "./service-with-tls --private-key=tls.key --certificate=tls.crt --ca-bundle=tls.ca"
+        cpus: 0.2
+        memory: 256
+        transport-encryption:
+          - name: tls
+            type: TLS
+        ports:
+            http:
+                protocol: tcp
+                port: 8080
+                vip:
+                    prefix: server-lb
+                    port: 80
+```
+
+Every item in the `transport-encryption` configuration section must have unique `name`. The `type` represents a format in which the private key and certificate will be delivered into the task sandbox. Different items will result in a unique private key and certificate.
```

Comment:
1st sentence: [...] must have *a* unique name.

2nd sentence: The `type` field defines the serialization format with which the private key and certificate files will be written into the task sandbox.

3rd sentence: Each item will result in a unique private key and the corresponding certificate.